action/staking/unstake.go

```diff
@@ -126,11 +127,19 @@ func runCheckUnstake(ctx *action.Context, tx action.RawTx) (bool, action.Respons
 		return false, action.Response{Log: err.Error()}
 	}
 
+	if ctx.EvidenceStore.IsFrozenValidator(ust.ValidatorAddress) {
+		return false, action.Response{Log: evidence.ErrFrozenValidator.Error()}
+	}
+
 	unstake := identity.Unstake{
 		Address: ust.ValidatorAddress,
 		Amount:  ust.Stake.Value,
 	}
 
+	if ctx.EvidenceStore.CheckRequestExists(ust.ValidatorAddress) {
```

Comment:
Is this the edge case that we disallow unstake when validator is under an on-going allegation?

Reply:
yes